pkg/trivyk8s/trivyk8s.go

```diff
@@ -0,0 +1,99 @@
+package trivyk8s
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/aquasecurity/trivy-kubernetes/pkg/artifacts"
+	"github.com/aquasecurity/trivy-kubernetes/pkg/k8s"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// GetKubeConfig returns local kubernetes configurations
+func GetKubeConfig() (*rest.Config, error) {
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+
+	kubeConfigPath := filepath.Join(userHomeDir, ".kube", "config")
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return kubeConfig, nil
+}
+
+// TrivyK8S interface represents the operations supported by the library
+type TrivyK8S interface {
+	ListArtifacts(context.Context, string) ([]*artifacts.Artifact, error)
```

Comment:
We might need another method for Iac resources. 
Iac scanner resources -  scan more resources than `workload` , such configmaps ,resourcequota ,randlimit and more.
Vulnerability scanner -  resources scan only workload resources
Basically, we need to fetch all supported resources for the namespace.
- for iac scanner  returns all resources
- for vulnerability scanner  return only workload resources

Reply:
I started with two separated methods, endup changing to one that returns everything because we were not sure if we wanted `trivy k8s` to support flags for specific scans (only iac, or only vulns). If `trivy k8s` will always do both scans at the same time and only have flags for `namespace`, or `cluster`, I think having only want method express better the intention of everything is a k8s scannable artifact, and trivy will run different scans on it.

Reply:
Regardless of the flags, resources are different for iac and vulnerability scanners. No reason for  vulnerability scanner to accept iac resources

Reply:
vulnerability scanner is not accepting it, it only scan images and resources do have images, and others don't.